backend/FwHeadless/Program.cs

```diff
@@ -113,6 +129,29 @@ static async Task<Results<Ok<SyncResult>, NotFound, ProblemHttpResult>> ExecuteM
     return TypedResults.Ok(result);
 }
 
+static async Task<Results<Ok<ProjectSyncStatus>, NotFound>> GetMergeStatus(
+    ProjectContext projectContext,
+    ProjectLookupService projectLookupService,
+    ProjectSyncStatusService syncStatusService,
+    IServiceProvider services,
+    LexBoxDbContext lexBoxDb,
+    Guid projectId)
+{
+    var status = syncStatusService.SyncStatus(projectId);
+    if (status is not null) return TypedResults.Ok(new ProjectSyncStatus(status.Value, 0));
+    var project = projectContext.Project;
+    if (project is null)
+    {
+        // 404 only means "project doesn't exist"; if we don't know the status, then it hasn't synced before and is therefore ready to sync
+        if (await projectLookupService.ProjectExists(projectId)) return TypedResults.Ok(ProjectSyncStatus.NeverSynced);
+        else return TypedResults.NotFound();
+    }
+    var commitsOnServer = await lexBoxDb.Set<ServerCommit>().CountAsync(c => c.ProjectId == projectId);
+    var lcmCrdtDbContext = services.GetRequiredService<LcmCrdtDbContext>(); // TODO: This *cannot* be right, can it?
+    var localCommits = await lcmCrdtDbContext.Set<Commit>().CountAsync();
```

Comment:
this is right that's all you need to do, that's the whole reason we set the context in the middleware is so that we can just get the service here

Reply:
I've removed the comment now, and I also inverted the direction of the subtraction so now it returns the right number. (3 commits pending, rather than negative 3 :grin:)